asm-lsp/lsp.rs

```diff
@@ -1666,33 +1674,30 @@ fn get_project_root(params: &InitializeParams) -> Option<PathBuf> {
     if let Some(folders) = &params.workspace_folders {
         // if there's multiple, just visit in order until we find a valid folder
         for folder in folders {
-            if let Ok(parsed) = PathBuf::from_str(folder.uri.path().as_str()) {
-                if let Ok(parsed_path) = parsed.canonicalize() {
-                    info!("Detected project root: {}", parsed_path.display());
-                    return Some(parsed_path);
-                }
+            let Ok(parsed) = PathBuf::from_str(folder.uri.path().as_str());
+            if let Ok(parsed_path) = parsed.canonicalize() {
+                info!("Detected project root: {}", parsed_path.display());
+                return Some(parsed_path);
             }
         }
     }
 
     // if workspace folders weren't set or came up empty, we check the root_uri
     #[allow(deprecated)]
     if let Some(root_uri) = &params.root_uri {
-        if let Ok(parsed) = PathBuf::from_str(root_uri.path().as_str()) {
-            if let Ok(parsed_path) = parsed.canonicalize() {
-                info!("Detected project root: {}", parsed_path.display());
-                return Some(parsed_path);
-            }
+        let Ok(parsed) = PathBuf::from_str(root_uri.path().as_str());
```

Comment:
This (and the other `irrefutable_let_patterns` lint fixes) builds on nightly but unfortunately doesn't yet on stable. Until that lands, do you think it would be better to throw on a `#[allow(irrefutable_let_patterns)]`, or use something like the following to make it clearer this is a temporary fix?

```rust
#[allow(irrefutable_let_patterns)]
let Ok(parsed) = PathBuf::from_str(folder.uri.path().as_str()) else {
unreachable!()
};
```

Reply:
Okay will patch it up, I would test with stable also before finalizing this PR.

Reply:
Running stable clippy on workspace and building on stable completed without warnings or errors, How did you build it to get those warnings/errors 🤔? Could you also verify you are on the latest stable?

Reply:
Funny timing! I tested this yesterday using `1.81`, which is what gave me the warnings/ errors. `1.82` released today, and checking the [release notes](https://blog.rust-lang.org/2024/10/17/Rust-1.82.0.html#omitting-empty-types-in-pattern-matching) it looks like they just stabilized this very feature! I think I'll have to update the version we're using in CI, but that *should* be pretty painless.